src/cirq_iqm/iqm_client.py

```diff
@@ -0,0 +1,179 @@
+    Status of a task
+    """
+    PENDING = "pending"
+    READY = "ready"
+    FAILED = "failed"
+
+
+@dataclass(frozen=True)
+class IQMInstruction:
+    """
+    Transfer DTO for IQM insrtructions
+    """
+    name: str
+    qubits: list[str]
+    args: dict
+
+
+@dataclass(frozen=True)
+class IQMCircuit:
+    """
+    Transfer DTO for IQM circuit
+    """
+    name: str
+    args: dict
+    instructions: list[IQMInstruction]
+
+
+@dataclass(frozen=True)
+class QubitMapping:
+    """
+    Mapping of logical qubits to physical qubits
+    """
+    logical_name: str
+    physical_name: str
+
+
+@dataclass(frozen=True)
+class RunResult:
+    """
+    Result of a task execution
+    """
+    status: RunStatus
+    measurements: dict[str:list[list]] = None
+    message: str = None
+
+    @staticmethod
+    def parse(inp: dict):
+        """
+        Parses the result from a dict
+        Args:
+            inp: value to parse
+
+        Returns:
+            Parsed object of RunResult
+
+        """
+        input_copy = inp.copy()
+        return RunResult(status=RunStatus(input_copy.pop("status")), **input_copy)
+
+
+class IQMBackendClient:
+    """
+    Class to access backend quantum computer
+    """
+    def __init__(self, url: str, token: str):
+        """
+        Init
+        Args:
+            url: Endpoint for accessing the quantum computer
+            token: Access token for authentification
+        """
+        self._token = token
+        self._base_url = url
+
+    def submit_circuit(self, circuit: IQMCircuit, mappings: list[QubitMapping] = None, shots: int = 1) -> int:
+        """
+        Submits circuit to the IQM backend
+        Args:
+            circuit: Circuit to be executed on the IQM backend
+            mappings: Mappings of human-readable names to physical names
+            shots: number of repetitions
+
+        Returns:
+            ID for the created task. This ID is needed to query the status and the execution results
+
+        """
+        result = requests.post(f"{self._base_url}/circuit/run", data={
+            "mappings": mappings,
+            "circuit": circuit,
+            "shots": shots
+        })
+        result.raise_for_status()
+        return json.loads(result.text)["id"]
+
+    def get_run(self, run_id) -> RunResult:
+        """
+        Query the status of the running task
+        Args:
+            run_id: id of the taks
+
+        Returns:
+            Run result (can be Pending)
+
+        Raises:
+            HTTPException for http exceptions
+            IQMException for IQM backend specific exceptions
+
+        """
+        result = requests.get(f"{self._base_url}/circuit/run/{run_id}")
+        result.raise_for_status()
+        result = RunResult.parse(json.loads(result.text))
+        if result.status == RunStatus.FAILED:
+            raise IQMException(result.message)
+        return result
+
+    def wait_results(self, run_id, timeout_secs=TIMEOUT_SECONDS) -> RunResult:
```

Comment:
`wait_for_results` would be more idiomatic.

Reply:
fixed